services/catalog/src/simcore_service_catalog/core/events.py

```diff
@@ -1,65 +1,90 @@
 import logging
-from collections.abc import Awaitable, Callable
-from typing import TypeAlias
+from collections.abc import AsyncIterator
 
 from fastapi import FastAPI
-from servicelib.fastapi.db_asyncpg_engine import close_db_connection, connect_to_db
-from servicelib.logging_utils import log_context
-
-from .._meta import APP_FINISHED_BANNER_MSG, APP_STARTED_BANNER_MSG
-from ..db.events import setup_default_product
-from ..services.director import close_director, setup_director
-from .background_tasks import start_registry_sync_task, stop_registry_sync_task
+from fastapi_lifespan_manager import LifespanManager, State
+from servicelib.fastapi.postgres_lifespan import (
+    PostgresLifespanStateKeys,
+    postgres_lifespan_manager,
+)
+from servicelib.fastapi.prometheus_instrumentation import (
+    lifespan_prometheus_instrumentation,
+)
+from servicelib.fastapi.tracing import initialize_tracing
+
+from .._meta import APP_FINISHED_BANNER_MSG, APP_NAME, APP_STARTED_BANNER_MSG
+from ..api.rpc.routes import setup_rpc_api_routes
+from ..db.events import setup_database
+from ..services.director import setup_director
+from ..services.function_services import setup_function_services
+from ..services.rabbitmq import setup_rabbitmq
+from .background_tasks import setup_background_task
+from .settings import ApplicationSettings
 
 _logger = logging.getLogger(__name__)
 
 
-EventCallable: TypeAlias = Callable[[], Awaitable[None]]
-
-
-def _flush_started_banner() -> None:
+def flush_started_banner() -> None:
     # WARNING: this function is spied in the tests
     print(APP_STARTED_BANNER_MSG, flush=True)  # noqa: T201
 
 
-def _flush_finished_banner() -> None:
+def flush_finished_banner() -> None:
     print(APP_FINISHED_BANNER_MSG, flush=True)  # noqa: T201
 
 
-def create_on_startup(app: FastAPI) -> EventCallable:
-    async def _() -> None:
-        _flush_started_banner()
+async def _setup_app(app: FastAPI) -> AsyncIterator[State]:
+    flush_started_banner()
+
+    settings: ApplicationSettings = app.state.settings
+
+    if settings.CATALOG_TRACING:
+        initialize_tracing(app, settings.CATALOG_TRACING, APP_NAME)
+
+    yield {
+        PostgresLifespanStateKeys.POSTGRES_SETTINGS: settings.CATALOG_POSTGRES,
+        "prometheus_instrumentation_enabled": settings.CATALOG_PROMETHEUS_INSTRUMENTATION_ENABLED,
+    }
+
+    flush_finished_banner()
+
+
+async def _setup_prometheus_instrumentation_adapter(
```

Comment:
I have a bit of an issue with the names here. After the usage of `lifespan` setup is split into `initialize` and `lifespan`. Some actions can no longer be taken into the lifespan, like you commented on your PR (eg middleware), I called those `initiaze` since they are not setup nor lifespan.
I would like to avoid mixing these names since they will give us a bad time.
If you don't like the names, please rename it everywhere it is but keep the difference. For sure I'd like to avoid having lifespan functions including setup in their names.
Please have a look at the `dynamic-scheduler` where this pattern was enforced. https://github.com/ITISFoundation/osparc-simcore/pull/7149